Modules/FindESMF.cmake

```diff
@@ -115,6 +115,17 @@ else()
 
 endif()
 
+# Add target aliases to avoid spurious '-l' flags.
+if(NOT TARGET ESMF::ESMF)
+  add_library(ESMF::ESMF ALIAS ESMF)
+endif()
+# The following 'ESMF' target alias can be removed after all UFS component CMake
+# configs transition to using 'ESMF::ESMF'. As it is, it creates backward
+# compatibility with ESMF/MAPL configurations that use the 'ESMF' target.
+if(NOT TARGET ESMF)
+  add_library(ESMF ALIAS ESMF)
+endif()
```

Comment:
I am not sure I understand this bit.
Why is this necessary and why is this after the aliasing on 120

Reply:
The 'ESMF' target is needed because that's what the MAPL config (MAPL-targets.cmake) is currently passing in to the GOCART CMake config. When this file doesn't define the capitalized 'ESMF' target, when GOCART essentially does `target_link_libraries(... ESMF)`, CMake doesn't find a target called 'ESMF', so it creates a flag '-lESMF' instead, which is not the behavior we want. So for the moment, this is the most important change.

The reason for the _other_ alias is that ESMF+MAPL will soon provide a 'ESMF::ESMF' target which will unambiguously tell CMake that we want a target and not a '-l' flag if the target for some reason couldn't be found. That said, those changes aren't in yet, so we could hold off on this piece. That said, all three targets ('esmf' and these two new aliases) can peacefully coexist.

Reply:
Thanks for explaining @AlexanderRichert-NOAA 
I need to digest that a bit, but won't hold this changeset for that.